### PR TITLE
service/fs: implement obsoleted Create/DeleteExtSaveData

### LINKED ISSUE #3537: Pokédex 3D failing to create Extra SD Card Data.
I've tried launching Pokédex 3D. The "game" seem to boot up correctly, but I can't get past the "Extra data will be created." screen, making the game unplayable. I'm using the latest Canary (HEAD-04273de).

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
Fix #3537 
This is quite easy to implement so I tried to do it, and it seemed like it goes well. Not sure if there are any issues though.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3948)

### CODE REVIEW

File: src/core/hle/service/fs/fs_user.h [hunk 1]
```diff
@@ -447,6 +447,34 @@ class FS_USER final : public ServiceFramework<FS_USER> {
      */
     void GetProgramLaunchInfo(Kernel::HLERequestContext& ctx);
 
+    /**
+     * FS_User::Obsoleted 3 0 CreateExtSaveData service function.
```
Comment (wwylele) [inline src/core/hle/service/fs/fs_user.h]: `Obsoleted_3_0_CreateExtSaveData`. 3dbrew's wiki doesn't correctly parse its title from url. Or you can just write `ObsoletedCreateExtSaveData` here.

File: src/core/hle/service/fs/fs_user.h [hunk 2]
```diff
@@ -447,6 +447,34 @@ class FS_USER final : public ServiceFramework<FS_USER> {
...
+     *      6 : Number of files
+     *      7 : (SMDH Size << 4) | 0x0000000A
+     *      8 : Pointer to the SMDH icon for the new ExtSaveData
+     *  Outputs:
+     *      1 : Result of function, 0 on success, otherwise error code
+     */
+    void ObsoletedCreateExtSaveData(Kernel::HLERequestContext& ctx);
+
+    /**
+     * FS_User::Obsoleted 3 0 DeleteExtSaveData service function.
```
Comment (wwylele) [inline src/core/hle/service/fs/fs_user.h]: ditto

**Comment by zhaowenlan1779** [comment]
@wwylele addressed

**Comment by zhaowenlan1779** [comment]
@Subv
fixes #3537 (Pokedex 3D)

**Pull request merged**